engine/execution/computation/computer/computer.go

```diff
@@ -220,23 +190,21 @@ func (e *blockComputer) executeBlock(
 
 	// executing system chunk
 	colView := stateView.NewChild()
-	systemCol, err := e.executeSystemCollection(blockSpan, collectionIndex, txIndex, systemChunkCtx, colView, res)
+	systemCol, err := e.executeSystemCollection(blockSpan, blockIdStr, collectionIndex, txIndex, systemChunkCtx, colView, collector)
 	if err != nil {
 		return nil, fmt.Errorf("failed to execute system chunk transaction: %w", err)
 	}
 
-	bc.Commit(colView)
-	eh.Hash(res.Events[len(res.Events)-1])
 	// close the views and wait for all views to be committed
-	bc.Close()
-	eh.Close()
+	collector.Close()
 
 	err = e.mergeView(stateView, colView, blockSpan, trace.EXEMergeCollectionView)
 	if err != nil {
 		return nil, fmt.Errorf("cannot merge view: %w", err)
 	}
 
-	wg.Wait()
+	collector.Wait()
+	res := collector.result
```

Comment:
does it make sense for Wait to return the result?

Reply:
that's reasonable.  I was thinking of combining Close/Wait/return result into a Finalize method anyways.